### PR TITLE
Add blurb about backwards compatability to upgrading-sdk-version docs

### PR DESCRIPTION
**Pull request by jberkhahn**
for #5032

It's been a while since we talked about this, so I hope this encapsulates everything we discussed about this.

### CODE REVIEW

**Comment by jmrodri** [comment]
@jberkhahn you forgot to sign your commit.

**Comment by jberkhahn** [comment]
woops, fixed it

File: website/content/en/docs/upgrading-sdk-version/_index.md [hunk 1]
```diff
@@ -3,3 +3,9 @@ title: Upgrade SDK Version
 weight: 4
 description: Guide to upgrading sdk version for your operator
 ---
+
+## Backwards Compatibility when Upgrading Operator-sdk version
+
+When upgrading your version of Operator-sdk, it is intended that post-1.0.0 minor versions (i.e. 1.y) are backwards compatible and strictly additive. Therefore, you
+only need to re-scaffold your operator with a newer version of Operator-sdk if you wish to take advantage of new features. If you do not wish to use new features,
```
Comment (asmacdo) [inline website/content/en/docs/upgrading-sdk-version/_index.md:10]:
```suggestion
only need to re-scaffold your operator with a newer version of Operator-SDK if you wish to take advantage of new features. If you do not wish to use new features,
```

**Review by asmacdo — APPROVED** [review]
One optional nit

/lgtm

**Comment by jberkhahn** [comment]
@asmacdo updated

**Comment by asmacdo** [comment]
Nit (dont fix, just next time) if you include `Closes #5032` in the commit message it will connect the issue to the PR

**Pull request merged**